src/iam/iam-base.ts

```diff
@@ -337,12 +337,25 @@ export class IAMBase {
 
   private async setupNATS() {
     if (this._natsServerUrl) {
-      let protocol = "ws";
-      if (this._natsServerUrl.indexOf("https://") === 0) {
-        protocol = "wss";
+      try {
+        let protocol = "ws";
+        if (this._natsServerUrl.indexOf("https://") === 0) {
+          protocol = "wss";
+        }
+
+        const connection = await Promise.race<NatsConnection | undefined>([
+          connect({
+            servers: `${protocol}://${this._natsServerUrl}`,
+            timeout: 2000
```

Comment:
Minor point, but I would consider assigning `2000` to a variable so that both timeouts can be sure to use the same thing